examples/jsm/libs/lottie_canvas.module.js

```diff
@@ -14841,4 +14843,8 @@ setExpressionsPlugin(Expressions);
 initialize$1();
 initialize();
 
-export { lottie as default };
+return lottie;
+
+} );
+
+export default lottie;
```

Comment:
from my tests, this optimization and most of the optimizations inside `examples/jsm/libs` are not needed

Reply:
After further review I have to affirm that they are necessary and precisely for the issues I outlined in the PR description. Rather than individually resolve these like I did in #26935, I thought it better to wrap their scopes in an IIFE for the sake of review.

Reply:
Not sure how you're testing, but I've made a repro branch of what I mean:
https://github.com/marcofugaro/treeshake-test/tree/addons-test

As you can see `LottieLoader` and `lottie_canvas.module.js` are not tree-shaken already.
I also tested `fflate.module.js`, `opentype.module.js`, `mmdparser.module.js` but I believe most of `jsm/libs` are already tree-shaken.

Reply:
As a matter of facts even other addons are tree-shaken already, I think having the proxy `Addons.js` is enough to make all addons tree-shakeable.
I tested with this

```js
// Addons.js
export * from './loaders/LottieLoader.js';
export * from './loaders/TTFLoader.js';
export * from './loaders/TiltLoader.js';
export * from './loaders/USDZLoader.js';
export * from './loaders/VOXLoader.js';
export * from './loaders/VRMLLoader.js';
export * from './loaders/VTKLoader.js';
export * from './loaders/XYZLoader.js';
export * from './loaders/lwo/IFFParser.js';
export * from './loaders/lwo/LWO2Parser.js';
export * from './loaders/lwo/LWO3Parser.js';
```

```
// index.js
import { LottieLoader } from 'three/addons'

console.log(LottieLoader)
```

And they all were already tree-shaken except LottieLoader (I am not testing this PR, but default three). But maybe I'm wrong, could you test this?

Reply:
I can provide repro of this if needed.

Reply:
As per my last message, I explained the aforementioned issues and testing strategy in the PR description. I'm not testing any optimizers like Uglify or Terser (Vite optionally uses the latter), nor am I treating three as external in the event that `package.sideEffects` is interactive. Please understand that I'm regarding the lowest common denominator between minification strategies for creating a baseline for testing and the sake of observability, so even though tools like ESBuild and Rollup have access to package information for smarter traversal, others only work at the module level and can't afford such optimizations.

Reply:
Yeah I did read the description. My point is that in the real world, those issues don't prevent the tree-shaking of #26910.

Unlike the main bundle, we're not importing from a single file here. Since the `package.sideEffects` flag is present, the bundlers are able to completely discard the unused files, even if they have some side-effects in it.

I did a test with the full Addons.js from #26910 and as you can see Esbuild, Rollup and Webpack can successfully tree-shake the Addons.js file, even without minification:
https://github.com/marcofugaro/treeshake-test/blob/addons-test-full/dist/index-esbuild.js

My point is, the changes in `jsm/libs` and the IIFE wraps make maintaning the repo more difficult, while not providing any real improvement to tree-shaking.